iree/compiler/Conversion/LinalgToLLVM/KernelDispatch.cpp

```diff
@@ -68,187 +70,166 @@ static llvm::cl::opt<int> genericOpsWorkgroupTileSize(
-          auto iteratorType = linalgOp.iterator_types()[i];
-          if (iteratorType.cast<StringAttr>().getValue() ==
-              getParallelIteratorTypeName()) {
-            workgroupTileSizes.push_back(genericOpsWorkgroupTileSize);
-          } else {
-            // Don't tile workgroup across reduction dimensions.
-            workgroupTileSizes.push_back(0);
-          }
-        }
-        return workgroupTileSizes;
-      }
-      // TODO(ataei): Set the parameters when we enable vectorization.
-      case TilingLevel::Level1Tiles:
-      case TilingLevel::Level2Tiles:
-        return {1, 1, 1};
-    }
+  if (contractionOp.isRowMajorBatchMatmul()) {
+    // TODO(ataei, ravishankarm): This should just use the configuration for
+    // matmul above. setting the tile size to 1 for all the batch dimensions.
+    TileSizesListType tileSizes = {
+        {1, batchMatmulWorkgroupTileSize, batchMatmulWorkgroupTileSize},
+        {1, batchMatmulL1TileSize, batchMatmulL1TileSize},
+        {1, batchMatmulL2TileSize, batchMatmulL2TileSize}};
+    SmallVector<int64_t, 4> nativeVectorSize = {1, batchMatmulL2TileSize,
+                                                batchMatmulL2TileSize};
+    IREE::HAL::LoweringConfig config =
+        getConfigAttr(tileSizes, nativeVectorSize, contractionOp->getContext());
+    setLoweringConfig(contractionOp, config);
+    return success();
   }
-
-  return {1, 1, 1};
+  return failure();
 }
-}  // namespace
-
-#define DEFINE_TILE_SIZE_FN(tilingLevel)                                      \
-  template <>                                                                 \
-  SmallVector<Value, 4> TileSizeFn::get<tilingLevel>(OpBuilder & builder,     \
-                                                     Operation * operation) { \
-    auto tileSizes = getTileSizes<tilingLevel>(operation);                    \
-    if (tileSizes.empty()) return {};                                         \
-    SmallVector<Value, 4> tileSizesVal;                                       \
-    tileSizesVal.reserve(tileSizes.size());                                   \
-    for (auto val : tileSizes) {                                              \
-      tileSizesVal.push_back(                                                 \
-          builder.create<ConstantIndexOp>(operation->getLoc(), val));         \
-    }                                                                         \
-    return tileSizesVal;                                                      \
+
+/// Sets the lowering configuration for dispatch region with root op being a
+/// generic op.
+LogicalResult setRootConfig(linalg::GenericOp genericOp) {
+  int64_t numOuterParallelLoops = getNumOuterParallelLoops(genericOp);
+  SmallVector<int64_t, 4> workgroupTileSizes(numOuterParallelLoops,
+                                             genericOpsWorkgroupTileSize);
+  SmallVector<int64_t, 4> innerTileSizes(numOuterParallelLoops, 1);
```

Comment:
Nit: Lets keep `TilingLevel` enum its more readable to keep `TilingLevel::WorkgroupTiles, TilingLevel::L1Tiles, TilingLevel::VectorTiles`

Reply:
I dont know how to fit in the enum in this way of doing this.

Reply:
They can still be local to the `KernelDispatch.h` so in `TileAndVectorizeLinalg` pass will do `getTileSizes(builder, operation, KernelDispatch::TileLevel:L1Tiles)` instead of `getTileSizes(builder, operation, 1)`

Reply:
That kind of tiling enum works for CPU, but can be misleading for other architectures. Having an enum where the value of the enum is load-bearing is something I try to avoid. Adding a new enum without thinking about how the enum value is used can be error-prone. I would prefer to leave it as an actual number. The attribute is just capturing the values for each level without takign opinions on what each level is for.

Reply:
Sure, the shared API will be enum free just the same way it is now:
```c++
getTileSizes(IREE::HAL::LoweringConfig config, unsigned level);
```

The enum is locally used within the CPU passes so you can do:
```c++
getTileSizes(config, KernelDispatch::TileLevel:VectorTiles)
```
Instead of :

```c++
getTileSizes(config, 2)
```

Reply:
Ok, I added that, but not a fan of making the value for the enum load-bearing. To me thats a potential footgun.